### PR TITLE
Add warning if variables are used twice

### PR DESCRIPTION
**Pull request by webrouse**
Part of: https://keboola.atlassian.net/browse/COM-1137

### CODE REVIEW

File: test/pull-variables-used-twice/expected-stderr [hunk 1]
```diff
@@ -0,0 +1 @@
+Warning: config "branch:%s/component:keboola.variables/config:%s" have 2 "variablesFor" relations [config "branch:%s/component:ex-generic-v2/config:%s"; config "branch:%s/component:ex-generic-v2/config:%s"], but only one is expected
```
Comment (webrouse) [inline test/pull-variables-used-twice/expected-stderr:1]: Funkcionalny test pre dalsi edge case:
- moze sa stat (API to nevaliduje), ze 2 konfiguracie obsahuju rovnake `variablesId`
- moze to nastat napr. pri klonovani konfiguracie v UI (nie je to atomicke)
- ak sa to stane, tak nevieme pod ktoru konfiguraciu `variables` ulozit
- v tom pripade vypiseme `warning`, a `variables` ulozime ako klasicku konfiguraciu, do samostatneho adresara -> tj. `Relation` sa ignoruje

File: test/pull-variables-used-twice/expected-stdout [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Plan for "pull" operation:
+  +  B main
+  +  C main/extractor/ex-generic-v2/with-variables-1
+  +  C main/extractor/ex-generic-v2/with-variables-2
+  +  C main/other/keboola.variables/variables
```
Comment (webrouse) [inline test/pull-variables-used-twice/expected-stdout:5]: `variables` su v samostatnom pricinku

File: internal/pkg/model/relation.go [hunk 3]
```diff
@@ -46,6 +47,24 @@ func (v Relations) GetByType(t RelationType) Relations {
...
 
+func (v Relations) GetAllByType() map[RelationType]Relations {
+	out := make(map[RelationType]Relations)
+	for _, relation := range v {
+		out[relation.Type()] = append(out[relation.Type()], relation)
+	}
+	return out
+}
+
+func (v *Relations) Add(relation Relation) {
```
Comment (webrouse) [inline internal/pkg/model/relation.go:66]: Nova metoda `Add ` - prida `Relation` do `Relations`, iba ak dana hodnota este nebola pridana.

**Review by JakubMatejka — APPROVED** [review]
